catalog/spatial/geocoding/spatial-geocoding-offline-catalog/src/main/java/org/codice/ddf/spatial/geocoding/query/GeoNamesQueryLuceneDirectoryIndex.java

```diff
@@ -121,11 +127,17 @@ static Shape getShape(String location) throws ParseException {
 
   @Override
   public Optional<String> getCountryCode(String wktLocation, int radius)
-      throws GeoEntryQueryException, ParseException {
-    final Directory directory = openDirectoryAndCheckForIndex();
+      throws GeoEntryQueryException {
+    String countryCode = null;
 
-    Shape shape = getShape(wktLocation);
-    String countryCode = doGetCountryCode(shape, radius, directory);
+    try (final Directory directory = openDirectoryAndCheckForIndex()) {
+      Shape shape = getShape(wktLocation);
+      countryCode = doGetCountryCode(shape, radius, directory);
+    } catch (IOException e) {
+      LOGGER.warn("Could not get directory");
```

Comment:
❓ Would it not be preferable to simply return Optional.empty() directly here and avoid creating a new Optional below with `null`?

Reply:
I thought about it. I like the method without the early returns out of the catch statements. I'm inclined to leave it as-is.

Reply:
Sure, it is just less optimal.